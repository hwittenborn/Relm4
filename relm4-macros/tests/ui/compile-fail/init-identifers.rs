use relm4::prelude::*;

struct ComponentInitBadIdentifiers;

#[relm4_macros::component]
impl SimpleComponent for ComponentInitBadIdentifiers {
    type Init = ();
    type Input = ();
    type Output = ();
    type Root = (i32, i32);

    view! {
        gtk::Window {}
    }

    fn init(
        _: Self::Init,
        (a, b): Self::Root,
        _: ComponentSender<Self>,
    ) -> ComponentParts<Self> {
        todo!();
    }

    fn init_root() -> Self::Root {
        todo!();
    }
}

struct ComponentInitNoArgs;

#[relm4_macros::component]
impl SimpleComponent for ComponentInitNoArgs {
    type Init = ();
    type Input = ();
    type Output = ();
    type Root = ();

    view! {
        gtk::Window {}
    }

    fn init() -> ComponentParts<Self> {
        todo!();
    }

    fn init_root() -> Self::Root {
        todo!();
    }
}

fn main() {}
