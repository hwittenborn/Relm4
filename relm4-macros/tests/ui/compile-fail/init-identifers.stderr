error: parameter binding must be an identifier
  --> tests/ui/compile-fail/init-identifers.rs:18:9
   |
18 |         (a, b): Self::Root,
   |         ^^^^^^

error: parameter binding must be an identifier
  --> tests/ui/compile-fail/init-identifers.rs:19:9
   |
19 |         _: ComponentSender<Self>,
   |         ^

error[E0050]: method `init` has 0 parameters but the declaration in trait `relm4::SimpleComponent::init` has 3
  --> tests/ui/compile-fail/init-identifers.rs:42:5
   |
42 |     fn init() -> ComponentParts<Self> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 parameters, found 0
   |
   = note: `init` from trait: `fn(<Self as relm4::SimpleComponent>::Init, <Self as relm4::SimpleComponent>::Root, relm4::ComponentSender<Self>) -> relm4::ComponentParts<Self>`
